frame/support/src/weights.rs

```diff
@@ -236,6 +247,35 @@ impl SimpleDispatchInfo {
 	}
 }
 
+/// A struct to represent a weight which is a function of the input arguments. The given closure's
+/// arguments must match the dispatch's argument types, given as a tuple.
```

Comment:
Maybe we could improve doc so the meaning of the different fields are directly understandable
```suggestion
/// arguments must match the dispatch's argument types, given as a tuple.
/// type is defined as: `FunctionOf(WeightDataFunction, ClassifyDispatchClass, PaysFeeBoolean)`
/// * WeightDataFunction is used to compute weight
/// * ClassifyDispatchClass is used to get the dispatch class
/// * PaysFeeBoolean is used to get whether it pays fees
```